lib/improver/tests/nowcasting/utilities/test_ApplyOrographicEnhancement.py

```diff
@@ -0,0 +1,584 @@
+                                [0., 1., MIN_PRECIP_RATE_MMH]]]])
+        expected1 = np.array([[[[MIN_PRECIP_RATE_MMH, MIN_PRECIP_RATE_MMH, 1.],
+                                [2., 3., 1.],
+                                [2., 3., 1.]]]])
+
+        # Mask values within the input precipitation cube that are equal to,
+        # or below 1.
+        precip_cubes = self.precip_cubes.copy()
+        new_precip_cubes = iris.cube.CubeList([])
+        for precip_cube in precip_cubes:
+            precip_cube.convert_units("mm/hr")
+            masked = np.ma.masked_where(
+                precip_cube.data <= 1, precip_cube.data)
+            precip_cube.data = masked
+            precip_cube.convert_units("m/s")
+            new_precip_cubes.append(precip_cube)
+
+        plugin = ApplyOrographicEnhancement("subtract")
+        result = plugin.process(self.precip_cubes, self.oe_cubes)
+        self.assertIsInstance(result, iris.cube.CubeList)
+        for aresult, precip_cube in zip(result, self.precip_cubes):
+            self.assertEqual(
+                aresult.metadata, precip_cube.metadata)
+        for cube in result:
+            cube.convert_units("mm/hr")
+        self.assertArrayAlmostEqual(result[0].data.data, expected0)
+        self.assertArrayAlmostEqual(result[1].data.data, expected1)
+
+    def test_one_input_cube(self):
+        """Test the addition of precipitation rate and orographic enhancement,
+        where a single precipitation rate cube is provided."""
+        expected = np.array([[[[0., 1., 2.],
+                               [1., 2., 7.],
+                               [0., 3., 4.]]]])
+        plugin = ApplyOrographicEnhancement("add")
+        result = plugin.process(self.precip_cubes[0], self.oe_cubes)
+        self.assertIsInstance(result, iris.cube.CubeList)
+        for aresult, precip_cube in zip(result, self.precip_cubes):
+            self.assertEqual(
+                aresult.metadata, precip_cube.metadata)
+        for cube in result:
+            cube.convert_units("mm/hr")
+        self.assertArrayAlmostEqual(result[0].data, expected)
+
+    def test_inputs_and_orographic_enhancements_as_cubes(self):
+        """Test the addition of precipitation rate and orographic enhancement,
+        where a single precipitation rate cube and a single orographic
+        enhancement cube is provided."""
+        expected = np.array([[[[0., 1., 2.],
+                               [1., 2., 7.],
+                               [0., 3., 4.]]]])
+        plugin = ApplyOrographicEnhancement("add")
+        result = plugin.process(self.precip_cubes[0], self.oe_cubes[0])
+        self.assertIsInstance(result, iris.cube.CubeList)
+        for aresult, precip_cube in zip(result, self.precip_cubes):
+            self.assertEqual(
+                aresult.metadata, precip_cube.metadata)
+        for cube in result:
+            cube.convert_units("mm/hr")
+        self.assertArrayAlmostEqual(result[0].data, expected)
+
+
```

Comment:
I want to be confident that repeated use of a single orographic enhancement field cannot be contaminated by the set-to-zero behaviour in one of the functions. Please can you add a test with two "time" points on the precip_cube that both match one "time" point on the orographic enhancement cube to satisfy this.
This won't be needed if you change the code in the _apply_orographic_enhancement function instead.

Reply:
I've added a copy of the cube, and a test for the specific method.